### ISSUE TITLE
Azure Backend ERROR CODE: AuthorizationPermissionMismatch

### ISSUE THREAD

**Issue by arthurderyckere-newday** [pulumi/pulumi#16587]
### What happened?

According to the [docs](https://www.pulumi.com/docs/concepts/state/#azure-blob-storage) Storage Blob Data Contributor should be enough to run pulumi login, but it doesn't work for us.

RESPONSE 403: 403 This request is not authorized to perform this operation using this permission.
ERROR CODE: AuthorizationPermissionMismatch

### Example

![role](https://github.com/pulumi/pulumi/assets/94851139/6815cccd-de4b-4771-a7d2-8262910fe4a7)

![image](https://github.com/pulumi/pulumi/assets/94851139/01024213-d24e-487d-a070-26e942fed013)

### Output of `pulumi about`

'dotnet build -nologo .' completed successfully
CLI          
Version      3.121.0
Go Version   go1.22.4
Go Compiler  gc

warning: Failed to get information about the current stack: No current stack
Plugins
KIND      NAME          VERSION
resource  azure-native  2.45.0
resource  azuread       5.52.0
language  dotnet        unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in dotnet: executable='/usr/bin/dotnet' version='8.0.[30](https://github.com/NewDayStratus/Stratus.AppRoleAssignments/actions/runs/9810821233/job/27091825103#step:6:31)0'

Backend        
Name           pulumi.com
URL            https://app.pulumi.com
User           Unknown
Organizations  
Token type     personal

No dependencies found

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by arthurderyckere-newday** [comment]
It looks like `pulumi login "azblob://${{inputs.pulumiContainer}}?storage_account${{inputs.pulumiStorageAccount}}"` uses DefaultAzureCredential which uses our self hosted GitHub runner managed identity instead of the one we want to use. How do we give the instruction to pulumi login to use a specific clientId to login?

**Comment by mikhailshilkov** [comment]
> uses DefaultAzureCredential which uses our self hosted GitHub runner managed identity instead of the one we want to use. How do we give the instruction to pulumi login with the AzureCliCredential?

Yeah, I think this is by design. We use gocloud.dev for blob storage, you can find its authentication options here: https://github.com/google/go-cloud/blob/0a37d2f8da8a73844216841be73381c92438d7e8/blob/azureblob/azureblob.go#L27

If `DefaultAzureCredential` MSI comes earlier than az login, so you either need to use one of gocloud's environment variables or bust MSI environment somehow.

**Comment by lunaris** [comment]
Closing this as it seems to have gone stale/been addressed as by design.
